tests/test_graph_processing.py

```diff
@@ -0,0 +1,116 @@
+"""
+Test file
+"""
+
+import pytest
+
+from power_system_simulation.graph_processing import (
+    GraphCycleError,
+    GraphNotFullyConnectedError,
+    GraphProcessor,
+    IDNotFoundError,
+    IDNotUniqueError,
+    InputLengthDoesNotMatchError,
+)
+
+
+class Tests:
+    """
+    Testing the raise exception in case of faulty graph
+    """
+
+    def test_id_not_unique_error_1(self):
+        """
```

Comment:
suggestion: Maybe you could make these test names a bit easier to distinguish
```suggestion
    def test_vertex_id_not_unique(self):
        """
```

Reply:
Same would go for the other tests, try describing in the test name what you are testing